### DISCUSSION TITLE
Github Action's Job Error: Process completed with exit code 1

### DISCUSSION THREAD

**Discussion by pawanJ09** [community/community#46905] [Actions] [answered]
### Select Topic Area

Bug

### Body

Getting error as Error: Process completed with exit code 1 in my Github workflow when I try to run a shell command. The aws lambda invoke command works but looks like the assignment to error_message variable might be failing.

Github Workflow log:
https://github.com/pawanJ09/expense-report-generator/actions/runs/4138750591/jobs/7155532322

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @pawanJ09 ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!

**Reply by pawanJ09** [reply]
Thanks @queenofcorgis for redirecting my query to the appropriate category !

**Comment by airtower-luna** [comment, marked as answer]
I assume the problem is that you pipe the `aws lambda invoke` output into `grep "errorMessage"`. That makes the exit status of the whole pipe that of `grep`, which will be non-zero (failure) if it can't find that text.

I'm not familiar with the AWS tool, so I don't know if its exit status might be enough to tell success or failure. If yes, that's the easiest option. If you need to parse the output, you could write it to a file and then `grep` from there (make sure to catch its exit code or you'll have the same problem).

**Reply by pawanJ09** [reply]
Thanks @airtower-luna, this was exactly the problem. I corrected my script to look for the correct tag and now the workflow works like a charm. 
Here is the updated code https://github.com/pawanJ09/expense-report-generator/blob/main/test.sh#L10 where I updated my error string from 'errorMessage' to 'FunctionError'
